design-toolkit/components/src/components/search-field/index.tsx

```diff
@@ -11,83 +11,121 @@
   Input,
+  composeRenderProps,
+  useContextProps,
 } from 'react-aria-components';
 import { Icon } from '../icon';
+import { SearchFieldStyles } from './styles';
+import type { SearchFieldProps } from './types';
+
+const { searchField, searchIcon, input, loadingIcon, clearButton } =
+  SearchFieldStyles();
+
+export const SearchFieldContext =
+  createContext<ContextValue<SearchFieldProps, HTMLDivElement>>(null);
 
-export interface SearchFieldProps {
-  className?: string;
-  /** Disables the input field. */
-  isDisabled?: boolean;
-  /** Displays a loading spinner. */
-  isLoading?: boolean;
-  /** Displays placeholder text. */
-  placeholder?: string;
-  /** Whether the input has a filled background or outlined. */
-  variant?: 'filled' | 'outlined';
+function SearchFieldProvider({
+  children,
+  ...props
+}: ProviderProps<SearchFieldProps>) {
+  return (
+    <SearchFieldContext.Provider value={props}>
+      {children}
+    </SearchFieldContext.Provider>
+  );
 }
+SearchFieldProvider.displayName = 'SearchField.Provider';
 
-const textFieldStyles = cva(
-  [
-    'hide-cancel block w-full rounded-round p-s pr-[30px] pl-[35px] font-display text-body-s outline',
-  ],
-  {
-    variants: {
-      isDisabled: {
-        true: 'text-disabled outline-interactive-disabled placeholder:text-disabled',
-        false:
-          'text-default-light placeholder:text-default-dark hover:outline-interactive-hover focus:outline-highlight',
-      },
-      variant: {
-        filled: 'bg-surface-raised outline-static-dark',
-        outlined: 'outline-interactive',
-      },
-    },
-  },
-);
+/**
+ * SearchField - A customizable search input component built on React Aria Components
+ *
+ * Provides a search input with integrated search icon, loading state, and clear functionality.
+ * Supports two visual variants (filled/outlined), and granular styling control.
+ *
+ * @example
+ * // Basic search field
+ * <SearchField placeholder="Search..." />
+ *
+ * @example
+ * // Filled variant with custom styling
+ * <SearchField
+ *   variant="filled"
+ *   placeholder="Search products"
+ *   classNames={{
+ *     input: "bg-gray-100",
+ *     searchIcon: "text-blue-600"
+ *   }}
+ * />
+ *
+ * @example
+ * // With event handlers
+ * <SearchField
+ *   placeholder="Type to search"
+ *   onSubmit={(value) => console.log('Search:', value)}
+ *   onChange={(value) => setQuery(value)}
+ * />
+ *
+ * @example
+ * // Using context provider for default props
+ * <SearchField.Provider variant="filled">
+ *   <SearchField placeholder="Search 1" />
+ *   <SearchField placeholder="Search 2" />
+ * </SearchField.Provider>
+ */
+export function SearchField({ ref, ...props }: SearchFieldProps) {
+  [props, ref] = useContextProps(props, ref ?? null, SearchFieldContext);
 
-export const SearchField = ({
-  className,
-  placeholder = 'Search',
-  variant = 'outlined',
-  isLoading = false,
-  ...rest
-}: SearchFieldProps) => {
+  const {
+    classNames,
+    placeholder,
```

Comment:
Can you use TextField as an example and exclude input props from the top level and instead implement `inputProps` as a prop that gets passed down and spread onto Input, please?

Reply:
I think I did it.